lib/scorecard.rb

```diff
@@ -0,0 +1,95 @@
+
+class ScoreCard
+  attr_reader :frames, :running_score, :frame_score, :scores
+
+  def initialize
+    @frames = 1
+    @running_score = []
+    @frame_score = []
+    @scores = []
+    @cumul_scores = []
+  end
+
+  def run
+    while @frames < 10 do
+      roll_1_input
+      if strike?
+        @frames += 1
+        clear_frame
+      else
+        roll_2_input
+        @frames += 1
+        clear_frame
+      end
+    end
+  roll_1_input
+  if strike?
+    roll_2_input
+    bonus_roll_input
+  else
+    roll_2_input
+  end
+  if spare?
+    bonus_roll_input
+  end
+  final_calc
+  puts "Your final score after 10 frames is #{@running_score.sum}"
+  cumul_calc
+  puts "Your cumulative score over the 10 frames is #{@cumul_scores}"
+  end
+
+  def roll_1_input
+    puts "Please enter roll 1 score"
+    roll_1 = gets.chomp.to_i
+    @scores << roll_1
+    @frame_score << roll_1
+  end
+
+  def roll_2_input
+    puts "Please enter roll 2 score"
+    roll_2 = gets.chomp.to_i
+    @scores << roll_2
+    @frame_score << roll_2
+  end
+
+  def bonus_roll_input
+    puts "Please enter roll 3 score"
+    bonus_roll = gets.chomp.to_i
+    @scores << bonus_roll
+    @frame_score << bonus_roll
+  end
+
+  def strike?
+    @frame_score[0] == 10
```

Comment:
10 is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) that appears throughout the code. We could refactor to have a clearly-named constant set to 10 to make the program more readable